src/rgw/rgw_pubsub_push.cc

```diff
@@ -141,6 +141,7 @@ class RGWPubSubHTTPEndpoint : public RGWPubSubEndpoint {
     const auto post_data = json_format_pubsub_event(event);
     request.set_post_data(post_data);
     request.set_send_length(post_data.length());
+    request.set_content_type("application/json");
```

Comment:
can't this just be:
```cpp
request.append_header("Content-Type", "application/json");
```
i'd rather not to add a separate interface specifically for content-type